File: ruby/beer-song/beer_song.rb
class BeerSong

  def verse(number)
    case number
    when 1
      i = "no more bottles"
      j = "#{number} bottle"
      k = "Take it down and pass it around"
    when 2
      i = "#{number - 1} bottle"
      j = "#{number} bottles"
      k = "Take one down and pass it around" 
    when 0
      i = "99 bottles"
      j = "No more bottles"
      k = "Go to the store and buy some more"
    else
      i = "#{number - 1} bottles"
      j = "#{number} bottles"
      k = "Take one down and pass it around" 
    end 

    "#{j} of beer on the wall, #{j.downcase} of beer.\n" \
    "#{k}, #{i} of beer on the wall.\n"
  end

  def verses(x, y)
    (y..x).to_a.reverse.map { |n| verse(n) }.join("\n") + "\n"
  end

  def sing
    verses(99, 0)
  end


end

File: javascript/beer-song/beer-song.js
var BeerSong = function(){};

BeerSong.prototype.verse = function(number){
  var i, j, k;
  switch(number){
    case 0:
      i = "No more bottles "
      j = "99 bottles "
      k = "Go to the store and buy some more"
      break;
    case 1:
      i = number +" bottle "
      j = "no more bottles "
      k = "Take it down and pass it around"
      break;
    case 2:
      i = number +" bottles "
      j = number-1 +" bottle "
      k = "Take one down and pass it around"
      break;
    default:
      i = number +" bottles "
      j = number-1 +" bottles "
      k = "Take one down and pass it around"
  }

  return i+"of beer on the wall, "+i.toLowerCase()+"of beer.\n"+k+", "+j+"of beer on the wall.\n"
}

BeerSong.prototype.sing = function(start, finish){
  finish = finish || 0;
  var verses = []
  for (var i = start; i >= finish; i--) {
    verses.push(this.verse(i));
  };
  return verses.join("\n");
}

module.exports = BeerSong;

